public/scripts/main.js

```diff
@@ -38,34 +38,46 @@ const keyToId = {
 // this tracks when we started asking for the current key command
 let questionStartMS = 0;
 
-$(document).ready(function() {
+$(document).ready(function () {
   //$("#retryButton").toggleClass("on");
   //alert($('li[data-keycode="test"]').attr('id'));
-   fetch('scripts/shortcuts.json')
-  .then(response => response.json())
-  .then(data => {
-    allData=data
-    if(localStorage.getItem("questionNo")==null)
-    {
-      localStorage.setItem("questionNo", "1");
-      localStorage.setItem("totalCount", Object.keys(allData).length);
-    }
-     readText()
-     updateTimingDisplay()
-  });
+  fetch('scripts/shortcuts.json')
+    .then(response => response.json()).then(data => {
+      allData = data
+      return getUserProgress()
+    }).then(response => {
+      if (!response.ok) {
+        const message = `An error has occured: ${response.status}`;
+        console.error(message)
+        return 1
+      }
+      return response.json()
+    })
+    .then(userProgress => {
+      if (localStorage.getItem("questionNo") == null) {
+        localStorage.setItem("questionNo", userProgress.currentQuestionNumber);
+        localStorage.setItem("totalCount", Object.keys(allData).length);
+      }
+      readText()
+      updateTimingDisplay()
+    });
 
   $('.container').css('height', $(window).height());
-  $(window).on('resize', function() {
+  $(window).on('resize', function () {
     $('.container').css('height', $(window).height());
   });
 });
 
+function getUserProgress() {
+  return fetch(document.URL + 'user/progress')
+}
+
 function nextQuestion() {
```

Comment:
I did auto formatting from this line to the last...Let me know if I need to remove this.

Reply:
Yes actually can you please only include changes relevant to the PR. Unfortunately we didn't get #89 merged prior to OSD which would've helped. Otherwise we'll get a bunch of spurious conflicts later.

If you're using VSCode, they added a really nice feature to only auto-format modified lines which you can enable in settings. Search settings for "Format". That really helps with these messy and non-standardized codebases 😄

If you're using another editor, one technique I've used is to do `git add -p` when staging which lets me just choose the hunks I've edited and not the rest of them produced by any auto-formatting. HTH